src/accessibility/RovingTabIndex.tsx

```diff
@@ -43,6 +43,22 @@ import { FocusHandler, Ref } from "./roving/types";
  * https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets#Technique_1_Roving_tabindex
  */
 
+// <input> types where it's safe to use the arrow keys to navigate over to the next item.
+// i.e. it's ok to press the down arrow on a radio button to move to the next radio.
+// But it's not ok to press the down arrow on a <input type="text"> to move away because
+// the down arrow should move the cursor to the end of the input.
+const NAVIGABLE_INPUT_TYPES = [
+    'radio',
+    'checkbox',
+];
+
+// Check for form elements which utilize the arrow keys for native functions
+// like many of the text input varieties.
+export function checkInputtableElement(el: HTMLElement): boolean {
+    const inputUsesArrowkeys = (el.tagName === 'INPUT' && !NAVIGABLE_INPUT_TYPES.includes(el.getAttribute('type')));
+    return inputUsesArrowkeys || el.matches("textarea, select, [contenteditable=true]");
```

Comment:
this feels like such a weird mix, we check for `input` using tagName but for `textarea` `select` using `.matches` - we could have this all be written as one selector.

`el.matches("input[type=radio], input[type=checkbox], textarea, select, [contenteditable=true]");`

Reply:
The smart move 😎 Good suggestion!

```js
el.matches("input:not([type=radio]):not([type=checkbox]), textarea, select, [contenteditable=true]");
```